packages/app-mobile/components/NoteEditor/NoteEditor.tsx

```diff
@@ -364,6 +364,23 @@ function NoteEditor(props: Props, ref: any) {
 		console.error('NoteEditor: webview error');
 	}, []);
 
+	const useMarkdownToolbar: boolean = useMemo(() => {
+		return Setting.value('editor.mobile.toolbarEnabled');
+	}, []);
+
```

Comment:
NoteEditor is a connected component, so you can map state.settings['editor.mobile.toolbarEnabled'] to props.something and use it here. It's the preferred pattern because it means if the settings change, the component correctly updates (although in this particular case it's not important, but it's still better to stick to the pattern).

Reply:
Oh that's right, I was looking at NoteEditor.tsx of the desktop app. How about passing it as prop from Note.tsx then? Since this one is connected. I prefer we don't connect too many components actually, except the top ones, since it means less global state, so it's good if NoteEditor for mobile is not connected

Reply:
There are lots of components that are not well implemented, but for new code we try to stick to new patterns. It means the day we need to refactor the component, there will be less code to refactor. So there's no need to deal with these 4 extra mappings for now.

And yes I want to avoid connecting more components as much as possible. I know not everyone agrees, but I feel this is global state and it makes it harder to see what happens and how. Not to mention reusability and testability is not great (although we don't test much for now).

Reply:
Changed!